packages/databricks-vscode/src/ui/bundle-resource-explorer/BundleCommands.ts

```diff
@@ -145,6 +147,44 @@ export class BundleCommands implements Disposable {
         this.bundleRunStatusManager.cancel(treeNode.resourceKey);
     }
 
+    @onError({popup: {prefix: "Error destroying bundle."}})
+    async destroy() {
+        if ((await this.configModel.get("mode")) !== "development") {
+            const confirm = await window.showErrorMessage(
+                "Are you sure you want to destroy this bundle and all resources associated with it?",
+                {modal: true},
+                "Yes, continue",
+                "No"
+            );
+
+            if (confirm !== "Yes, continue") {
+                return;
+            }
+        }
+
+        try {
+            this.whenContext.setDeploymentState("deploying");
+            await window.withProgress(
+                {location: ProgressLocation.Notification, cancellable: false},
+                async () => {
+                    await this.bundleRemoteStateModel.destroy();
+                }
+            );
+
+            await this.refreshRemoteState();
+        } catch (e) {
+            if (!(e instanceof Error)) {
+                throw e;
+            }
+            if (e instanceof ProcessError) {
+                e.showErrorMessage("Error destroying bundle.");
+            }
+            throw e;
```

Comment:
Wouldn't we show the error twice? From the `e.showErrorMessage` and the `@onError`

Reply:
Good catch. Fixed!